packages/medusa/src/api/middlewares/with-default-sales-channel.ts

```diff
@@ -0,0 +1,28 @@
+import { NextFunction, Request, Response } from "express"
+import { SalesChannelService } from "../../services"
+
+/**
+ * Middleware that includes the default sales channel on the request, if no sales channels present
+ * @param asArray Whether to include the default sales channel id as an array or just a string
+ */
+export function withDefaultSalesChannel(
+  asArray?: boolean
+): (req: Request, res: Response, next: NextFunction) => Promise<void> {
+  return async (req: Request, _, next: NextFunction) => {
+    if (req.query.sales_channel_id?.length) {
```

Comment:
todo: user can define sales channels for the request with a PublishableApiKey as well, should we check for the existence of SC if PK is present in the header?

Reply:
Ah yes, good shout, will add the check.

Reply:
+1 - nice catch